File: src/BigPicture/BigPicture.Resolver.CSharp/Nodes/CompileItem.cs
using BigPicture.Core;
using System;
using System.Collections.Generic;
using System.Text;

namespace BigPicture.Resolver.CSharp.Nodes
{
    public class CompileItem : Node
    {
        public String Path { get; set; }
        public String AbsolutePath { get; set; }
        public String SubType { get; set; }
        public String AutoGen { get; set; }
        public String DesignTime { get; set; }
        public String CodeBlockId { get; set; }
    }
}

File: src/big-picture-ui/src/model/c-sharp/node/compile-item.ts
import Node from '../../base/node';
import { CodeEditor } from '../../components/code-editor';
import NavigationItem, { NavigationItems } from '../../components/base/navigation-item';

export class CompileItem extends Node {
  Path: string;
  AbsolutePath: string;
  SubType: string;
  AutoGen: string;
  DesignTime: string;

  public getNavigationItems(): NavigationItem[]  {
    var superNavItems = super.getNavigationItems();
    return [
      ...superNavItems,
      NavigationItems.code
    ];
  }
}